diff --git a/BUILD b/BUILD
index 3b3abec..5f304be 100644
--- a/BUILD
+++ b/BUILD
@@ -1,6 +1,5 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library", "go_test")
 load("@bazel_gazelle//:def.bzl", "gazelle")
-load("@build_bazel_rules_nodejs//:index.bzl", "pkg_npm")
 
 # gazelle:prefix github.com/bazelbuild/bazelisk
 gazelle(name = "gazelle")
@@ -151,23 +150,3 @@ go_binary(
     pure = "on",
     visibility = ["//visibility:public"],
 )
-
-pkg_npm(
-    name = "npm_package",
-    package_name = "@bazel/bazelisk",
-    srcs = [
-        "LICENSE",
-        "README.md",
-        "bazelisk.d.ts",
-        "bazelisk.js",
-        "package.json",
-    ],
-    substitutions = {"0.0.0-PLACEHOLDER": "{BUILD_SCM_VERSION}"},
-    deps = [
-        ":bazelisk-darwin-amd64",
-        ":bazelisk-darwin-arm64",
-        ":bazelisk-linux-amd64",
-        ":bazelisk-linux-arm64",
-        ":bazelisk-windows-amd64",
-    ],
-)
