diff --git a/buildifier/BUILD.bazel b/buildifier/BUILD.bazel
index 860cc7e..327f7b5 100644
--- a/buildifier/BUILD.bazel
+++ b/buildifier/BUILD.bazel
@@ -76,7 +76,7 @@ go_library(
     name = "buildifier_lib",
     srcs = ["buildifier.go"],
     importpath = "github.com/bazelbuild/buildtools/buildifier",
-    visibility = ["//visibility:private"],
+    visibility = ["//visibility:public"],
     x_defs = {
         "main.buildVersion": "{STABLE_buildVersion}",
         "main.buildScmRevision": "{STABLE_buildScmRevision}",
diff --git a/buildifier/buildifier.go b/buildifier/buildifier.go
index 9d0a677..ff49cc0 100644
--- a/buildifier/buildifier.go
+++ b/buildifier/buildifier.go
@@ -95,6 +95,10 @@ with -config=off.
 }
 
 func main() {
+	os.Exit(Run())
+}
+
+func Run() int {
 	c := config.New()
 
 	flags := c.FlagSet("buildifier", flag.ExitOnError)
@@ -159,7 +163,7 @@ func main() {
 	b := buildifier{c, differ}
 	exitCode := b.run(args)
 
-	os.Exit(exitCode)
+	return exitCode
 }
 
 type buildifier struct {
