diff --git a/expfmt/BUILD.bazel b/expfmt/BUILD.bazel
index 22dd614..66a6621 100644
--- a/expfmt/BUILD.bazel
+++ b/expfmt/BUILD.bazel
@@ -16,7 +16,7 @@ go_library(
         "//internal/bitbucket.org/ww/goautoneg",
         "//model",
         "@com_github_matttproud_golang_protobuf_extensions//pbutil",
-        "@com_github_prometheus_client_model//go",
+        "@@//proto:prometheus_client_go_proto",
         "@org_golang_google_protobuf//encoding/prototext",
         "@org_golang_google_protobuf//proto",
     ],
@@ -43,7 +43,7 @@ go_test(
     deps = [
         "//model",
         "@com_github_matttproud_golang_protobuf_extensions//pbutil",
-        "@com_github_prometheus_client_model//go",
+        "@@//proto:prometheus_client_go_proto",
         "@org_golang_google_protobuf//proto",
         "@org_golang_google_protobuf//types/known/timestamppb",
     ],
