diff --git a/src/main/protobuf/BUILD.bazel b/src/main/protobuf/BUILD.bazel
index 5d81948..fac604b 100644
--- a/src/main/protobuf/BUILD.bazel
+++ b/src/main/protobuf/BUILD.bazel
@@ -199,7 +199,7 @@ proto_library(
 
 go_proto_library(
     name = "analysis_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/analysis_go_proto",
     proto = ":analysis_proto",
     visibility = ["//visibility:public"],
     deps = [":blaze_query_go_proto"],
@@ -207,7 +207,7 @@ go_proto_library(
 
 go_proto_library(
     name = "bazel_flags_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/bazel_flags_go_proto",
     proto = ":bazel_flags_proto",
     visibility = ["//visibility:public"],
 )
@@ -215,7 +215,7 @@ go_proto_library(
 go_proto_library(
     name = "bazel_output_service_go_proto",
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/bazel_output_service_go_proto",
     proto = ":bazel_output_service_proto",
     visibility = ["//visibility:public"],
     deps = ["//google/rpc:status_proto"],
@@ -223,7 +223,7 @@ go_proto_library(
 
 go_proto_library(
     name = "bazel_output_service_rev2_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/bazel_output_service_rev2_go_proto",
     proto = ":bazel_output_service_rev2_proto",
     visibility = ["//visibility:public"],
     deps = ["//build/bazel/remote/execution/v2:remote_execution_proto"],
@@ -231,21 +231,21 @@ go_proto_library(
 
 go_proto_library(
     name = "blaze_deps_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/blaze_deps_go_proto",
     proto = ":blaze_deps_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "blaze_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/blaze_go_proto",
     proto = ":blaze_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "blaze_query_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/blaze_query_go_proto",
     proto = ":blaze_query_proto",
     visibility = ["//visibility:public"],
     deps = [":stardoc_output_go_proto"],
@@ -253,21 +253,21 @@ go_proto_library(
 
 go_proto_library(
     name = "buildjar_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/buildjar_go_proto",
     proto = ":buildjar_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "builtin_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/builtin_go_proto",
     proto = ":builtin_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "command_line_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/command_line_go_proto",
     proto = ":command_line_proto",
     visibility = ["//visibility:public"],
     deps = [":options_go_proto"],
@@ -276,7 +276,7 @@ go_proto_library(
 go_proto_library(
     name = "command_server_go_proto",
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/command_server_go_proto",
     proto = ":command_server_proto",
     visibility = ["//visibility:public"],
     deps = [":failure_details_go_proto"],
@@ -284,42 +284,42 @@ go_proto_library(
 
 go_proto_library(
     name = "crash_debugging_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/crash_debugging_go_proto",
     proto = ":crash_debugging_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "crosstool_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/crosstool_go_proto",
     proto = ":crosstool_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "desugar_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/desugar_go_proto",
     proto = ":desugar_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "execution_graph_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/execution_graph_go_proto",
     proto = ":execution_graph_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "failure_details_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/failure_details_go_proto",
     proto = ":failure_details_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "invocation_policy_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/invocation_policy_go_proto",
     proto = ":invocation_policy_proto",
     visibility = ["//visibility:public"],
     deps = [":strategy_policy_go_proto"],
@@ -327,28 +327,28 @@ go_proto_library(
 
 go_proto_library(
     name = "memory_pressure_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/memory_pressure_go_proto",
     proto = ":memory_pressure_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "options_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/options_go_proto",
     proto = ":options_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "protos_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/protos_go_proto",
     proto = ":protos_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "remote_logging_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/remote_logging_go_proto",
     proto = ":remote_logging_proto",
     visibility = ["//visibility:public"],
     deps = [
@@ -361,42 +361,42 @@ go_proto_library(
 
 go_proto_library(
     name = "remote_scrubbing_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/remote_scrubbing_go_proto",
     proto = ":remote_scrubbing_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "skyframe_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/skyframe_go_proto",
     proto = ":skyframe_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "stardoc_output_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/stardoc_output_go_proto",
     proto = ":stardoc_output_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "strategy_policy_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/strategy_policy_go_proto",
     proto = ":strategy_policy_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "worker_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/worker_go_proto",
     proto = ":worker_proto",
     visibility = ["//visibility:public"],
 )
 
 go_proto_library(
     name = "xcode_config_go_proto",
-    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf/xcode_config_go_proto",
     proto = ":xcode_config_proto",
     visibility = ["//visibility:public"],
 )
