diff events/container.proto events/container.proto
--- events/container.proto
+++ events/container.proto
@@ -25,7 +25,7 @@ option go_package = "github.com/containerd/containerd/api/events;events";
 option (containerd.types.fieldpath_all) = true;
 
 message ContainerCreate {
-	string id = 1;
+	string i_d = 1;
 	string image = 2;
 	message Runtime {
 		string name = 1;
@@ -35,12 +35,12 @@ message ContainerCreate {
 }
 
 message ContainerUpdate {
-	string id = 1;
+	string i_d = 1;
 	string image = 2;
 	map<string, string> labels  = 3;
 	string snapshot_key = 4;
 }
 
 message ContainerDelete {
-	string id = 1;
+	string i_d = 1;
 }
diff events/sandbox.proto b/events/sandbox.proto
--- events/sandbox.proto
+++ events/sandbox.proto
@@ -23,15 +23,15 @@ import "google/protobuf/timestamp.proto";
 option go_package = "github.com/containerd/containerd/api/events;events";
 
 message SandboxCreate {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 }
 
 message SandboxStart {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 }
 
 message SandboxExit {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	uint32 exit_status = 2;
 	google.protobuf.Timestamp exited_at = 3;
 }
diff events/task.proto events/task.proto
--- events/task.proto
+++ events/task.proto
@@ -26,7 +26,7 @@ option go_package = "github.com/containerd/containerd/api/events;events";
 option (containerd.types.fieldpath_all) = true;
 
 message TaskCreate {
-	string container_id = 1;
+	string container_i_d = 1;
 	string bundle = 2;
 	repeated containerd.types.Mount rootfs = 3;
 	TaskIO io = 4;
@@ -35,18 +35,18 @@ message TaskCreate {
 }
 
 message TaskStart {
-	string container_id = 1;
+	string container_i_d = 1;
 	uint32 pid = 2;
 }
 
 message TaskDelete {
-	string container_id = 1;
+	string container_i_d = 1;
 	uint32 pid = 2;
 	uint32 exit_status = 3;
 	google.protobuf.Timestamp exited_at = 4;
 	// id is the specific exec. By default if omitted will be `""` thus matches
 	// the init exec of the task matching `container_id`.
-	string id = 5;
+	string i_d = 5;
 }
 
 message TaskIO {
@@ -57,37 +57,37 @@ message TaskIO {
 }
 
 message TaskExit {
-	string container_id = 1;
-	string id = 2;
+	string container_i_d = 1;
+	string i_d = 2;
 	uint32 pid = 3;
 	uint32 exit_status = 4;
 	google.protobuf.Timestamp exited_at = 5;
 }
 
 message TaskOOM {
-	string container_id = 1;
+	string container_i_d = 1;
 }
 
 message TaskExecAdded {
-	string container_id = 1;
-	string exec_id = 2;
+	string container_i_d = 1;
+	string exec_i_d = 2;
 }
 
 message TaskExecStarted {
-	string container_id = 1;
-	string exec_id = 2;
+	string container_i_d = 1;
+	string exec_i_d = 2;
 	uint32 pid = 3;
 }
 
 message TaskPaused {
-	string container_id = 1;
+	string container_i_d = 1;
 }
 
 message TaskResumed {
-	string container_id = 1;
+	string container_i_d = 1;
 }
 
 message TaskCheckpointed {
-	string container_id = 1;
+	string container_i_d = 1;
 	string checkpoint = 2;
 }
diff runtime/sandbox/v1/sandbox.proto runtime/sandbox/v1/sandbox.proto
--- runtime/sandbox/v1/sandbox.proto
+++ runtime/sandbox/v1/sandbox.proto
@@ -62,7 +62,7 @@ service Sandbox {
 }
 
 message CreateSandboxRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	string bundle_path = 2;
 	repeated containerd.types.Mount rootfs = 3;
 	google.protobuf.Any options = 4;
@@ -73,7 +73,7 @@ message CreateSandboxRequest {
 message CreateSandboxResponse {}
 
 message StartSandboxRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 }
 
 message StartSandboxResponse {
@@ -82,7 +82,7 @@ message StartSandboxResponse {
 }
 
 message PlatformRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 }
 
 message PlatformResponse {
@@ -90,20 +90,20 @@ message PlatformResponse {
 }
 
 message StopSandboxRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	uint32 timeout_secs = 2;
 }
 
 message StopSandboxResponse {}
 
 message UpdateSandboxRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	google.protobuf.Any resources = 2;
 	map<string, string> annotations = 3;
 }
 
 message WaitSandboxRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 }
 
 message WaitSandboxResponse {
@@ -114,12 +114,12 @@ message WaitSandboxResponse {
 message UpdateSandboxResponse {}
 
 message SandboxStatusRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	bool verbose = 2;
 }
 
 message SandboxStatusResponse {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	uint32 pid = 2;
 	string state = 3;
 	map<string, string> info = 4;
@@ -129,19 +129,19 @@ message SandboxStatusResponse {
 }
 
 message PingRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 }
 
 message PingResponse {}
 
 message ShutdownSandboxRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 }
 
 message ShutdownSandboxResponse {}
 
 message SandboxMetricsRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 }
 
 message SandboxMetricsResponse {
diff runtime/task/v2/shim.proto runtime/task/v2/shim.proto
--- runtime/task/v2/shim.proto
+++ runtime/task/v2/shim.proto
@@ -51,7 +51,7 @@ service Task {
 }
 
 message CreateTaskRequest {
-	string id = 1;
+	string i_d = 1;
 	string bundle = 2;
 	repeated containerd.types.Mount rootfs = 3;
 	bool terminal = 4;
@@ -68,8 +68,8 @@ message CreateTaskResponse {
 }
 
 message DeleteRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 }
 
 message DeleteResponse {
@@ -79,8 +79,8 @@ message DeleteResponse {
 }
 
 message ExecProcessRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 	bool terminal = 3;
 	string stdin = 4;
 	string stdout = 5;
@@ -92,19 +92,19 @@ message ExecProcessResponse {
 }
 
 message ResizePtyRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 	uint32 width = 3;
 	uint32 height = 4;
 }
 
 message StateRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 }
 
 message StateResponse {
-	string id = 1;
+	string i_d = 1;
 	string bundle = 2;
 	uint32 pid = 3;
 	containerd.v1.types.Status status = 4;
@@ -114,24 +114,24 @@ message StateResponse {
 	bool terminal = 8;
 	uint32 exit_status = 9;
 	google.protobuf.Timestamp exited_at = 10;
-	string exec_id = 11;
+	string exec_i_d = 11;
 }
 
 message KillRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 	uint32 signal = 3;
 	bool all = 4;
 }
 
 message CloseIORequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 	bool stdin = 3;
 }
 
 message PidsRequest {
-	string id = 1;
+	string i_d = 1;
 }
 
 message PidsResponse {
@@ -139,20 +139,20 @@ message PidsResponse {
 }
 
 message CheckpointTaskRequest {
-	string id = 1;
+	string i_d = 1;
 	string path = 2;
 	google.protobuf.Any options = 3;
 }
 
 message UpdateTaskRequest {
-	string id = 1;
+	string i_d = 1;
 	google.protobuf.Any resources = 2;
 	map<string, string> annotations = 3;
 }
 
 message StartRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 }
 
 message StartResponse {
@@ -160,8 +160,8 @@ message StartResponse {
 }
 
 message WaitRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 }
 
 message WaitResponse {
@@ -170,7 +170,7 @@ message WaitResponse {
 }
 
 message StatsRequest {
-	string id = 1;
+	string i_d = 1;
 }
 
 message StatsResponse {
@@ -178,7 +178,7 @@ message StatsResponse {
 }
 
 message ConnectRequest {
-	string id = 1;
+	string i_d = 1;
 }
 
 message ConnectResponse {
@@ -188,14 +188,14 @@ message ConnectResponse {
 }
 
 message ShutdownRequest {
-	string id = 1;
+	string i_d = 1;
 	bool now = 2;
 }
 
 message PauseRequest {
-	string id = 1;
+	string i_d = 1;
 }
 
 message ResumeRequest {
-	string id = 1;
+	string i_d = 1;
 }
diff runtime/task/v3/shim.proto runtime/task/v3/shim.proto
--- runtime/task/v3/shim.proto
+++ runtime/task/v3/shim.proto
@@ -51,7 +51,7 @@ service Task {
 }
 
 message CreateTaskRequest {
-	string id = 1;
+	string i_d = 1;
 	string bundle = 2;
 	repeated containerd.types.Mount rootfs = 3;
 	bool terminal = 4;
@@ -68,8 +68,8 @@ message CreateTaskResponse {
 }
 
 message DeleteRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 }
 
 message DeleteResponse {
@@ -79,8 +79,8 @@ message DeleteResponse {
 }
 
 message ExecProcessRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 	bool terminal = 3;
 	string stdin = 4;
 	string stdout = 5;
@@ -92,19 +92,19 @@ message ExecProcessResponse {
 }
 
 message ResizePtyRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 	uint32 width = 3;
 	uint32 height = 4;
 }
 
 message StateRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 }
 
 message StateResponse {
-	string id = 1;
+	string i_d = 1;
 	string bundle = 2;
 	uint32 pid = 3;
 	containerd.v1.types.Status status = 4;
@@ -114,24 +114,24 @@ message StateResponse {
 	bool terminal = 8;
 	uint32 exit_status = 9;
 	google.protobuf.Timestamp exited_at = 10;
-	string exec_id = 11;
+	string exec_i_d = 11;
 }
 
 message KillRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 	uint32 signal = 3;
 	bool all = 4;
 }
 
 message CloseIORequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 	bool stdin = 3;
 }
 
 message PidsRequest {
-	string id = 1;
+	string i_d = 1;
 }
 
 message PidsResponse {
@@ -139,20 +139,20 @@ message PidsResponse {
 }
 
 message CheckpointTaskRequest {
-	string id = 1;
+	string i_d = 1;
 	string path = 2;
 	google.protobuf.Any options = 3;
 }
 
 message UpdateTaskRequest {
-	string id = 1;
+	string i_d = 1;
 	google.protobuf.Any resources = 2;
 	map<string, string> annotations = 3;
 }
 
 message StartRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 }
 
 message StartResponse {
@@ -160,8 +160,8 @@ message StartResponse {
 }
 
 message WaitRequest {
-	string id = 1;
-	string exec_id = 2;
+	string i_d = 1;
+	string exec_i_d = 2;
 }
 
 message WaitResponse {
@@ -170,7 +170,7 @@ message WaitResponse {
 }
 
 message StatsRequest {
-	string id = 1;
+	string i_d = 1;
 }
 
 message StatsResponse {
@@ -178,7 +178,7 @@ message StatsResponse {
 }
 
 message ConnectRequest {
-	string id = 1;
+	string i_d = 1;
 }
 
 message ConnectResponse {
@@ -188,14 +188,14 @@ message ConnectResponse {
 }
 
 message ShutdownRequest {
-	string id = 1;
+	string i_d = 1;
 	bool now = 2;
 }
 
 message PauseRequest {
-	string id = 1;
+	string i_d = 1;
 }
 
 message ResumeRequest {
-	string id = 1;
+	string i_d = 1;
 }
diff services/containers/v1/containers.proto services/containers/v1/containers.proto
--- services/containers/v1/containers.proto
+++ services/containers/v1/containers.proto
@@ -54,7 +54,7 @@ message Container {
 	// ID is the user-specified identifier.
 	//
 	// This field may not be updated.
-	string id = 1;
+	string i_d = 1;
 
 	// Labels provides an area to include arbitrary data on containers.
 	//
@@ -119,7 +119,7 @@ message Container {
 }
 
 message GetContainerRequest {
-	string id = 1;
+	string i_d = 1;
 }
 
 message GetContainerResponse {
@@ -173,7 +173,7 @@ message UpdateContainerResponse {
 }
 
 message DeleteContainerRequest {
-	string id = 1;
+	string i_d = 1;
 }
 
 message ListContainerMessage {
diff services/introspection/v1/introspection.proto services/introspection/v1/introspection.proto
--- services/introspection/v1/introspection.proto
+++ services/introspection/v1/introspection.proto
@@ -47,7 +47,7 @@ message Plugin {
 	string type = 1;
 
 	// ID identifies the plugin uniquely in the system.
-	string id = 2;
+	string i_d = 2;
 
 	// Requires lists the plugin types required by this plugin.
 	repeated string requires = 3;
@@ -104,21 +104,21 @@ message PluginsResponse {
 }
 
 message ServerResponse {
-	string uuid = 1;
+	string u_u_i_d = 1;
 	uint64 pid = 2;
 	uint64 pidns = 3; // PID namespace, such as 4026531836
 	repeated DeprecationWarning deprecations = 4;
 }
 
 message DeprecationWarning {
-	string id = 1;
+	string i_d = 1;
 	string message = 2;
 	google.protobuf.Timestamp last_occurrence = 3;
 }
 
 message PluginInfoRequest {
 	string type = 1;
-	string id = 2;
+	string i_d = 2;
 
 	// Options may be used to request extra dynamic information from
 	// a plugin.
diff services/leases/v1/leases.proto services/leases/v1/leases.proto
--- services/leases/v1/leases.proto
+++ services/leases/v1/leases.proto
@@ -49,7 +49,7 @@ service Leases {
 
 // Lease is an object which retains resources while it exists.
 message Lease {
-	string id = 1;
+	string i_d = 1;
 
 	google.protobuf.Timestamp created_at = 2;
 
@@ -59,7 +59,7 @@ message Lease {
 message CreateRequest {
 	// ID is used to identity the lease, when the id is not set the service
 	// generates a random identifier for the lease.
-	string id = 1;
+	string i_d = 1;
 
 	map<string, string> labels = 3;
 }
@@ -69,7 +69,7 @@ message CreateResponse {
 }
 
 message DeleteRequest {
-	string id = 1;
+	string i_d = 1;
 
 	// Sync indicates that the delete and cleanup should be done
 	// synchronously before returning to the caller
@@ -87,7 +87,7 @@ message ListResponse {
 }
 
 message Resource {
-	string id = 1;
+	string i_d = 1;
 
 	// For snapshotter resource, there are many snapshotter types here, like
 	// overlayfs, devmapper etc. The type will be formatted with type,
@@ -96,19 +96,19 @@ message Resource {
 }
 
 message AddResourceRequest {
-	string id = 1;
+	string i_d = 1;
 
 	Resource resource = 2;
 }
 
 message DeleteResourceRequest {
-	string id = 1;
+	string i_d = 1;
 
 	Resource resource = 2;
 }
 
 message ListResourcesRequest {
-	string id = 1;
+	string i_d = 1;
 }
 
 message ListResourcesResponse {
diff services/sandbox/v1/sandbox.proto services/sandbox/v1/sandbox.proto
--- services/sandbox/v1/sandbox.proto
+++ services/sandbox/v1/sandbox.proto
@@ -64,7 +64,7 @@ message StoreUpdateResponse {
 }
 
 message StoreDeleteRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 }
 
 message StoreDeleteResponse {}
@@ -78,7 +78,7 @@ message StoreListResponse {
 }
 
 message StoreGetRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 }
 
 message StoreGetResponse {
@@ -99,7 +99,7 @@ service Controller {
 }
 
 message ControllerCreateRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	repeated containerd.types.Mount rootfs = 2;
 	google.protobuf.Any options = 3;
 	string netns_path = 4;
@@ -109,16 +109,16 @@ message ControllerCreateRequest {
 }
 
 message ControllerCreateResponse {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 }
 
 message ControllerStartRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	string sandboxer = 10;
 }
 
 message ControllerStartResponse {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	uint32 pid = 2;
 	google.protobuf.Timestamp created_at = 3;
 	map<string, string> labels = 4;
@@ -130,7 +130,7 @@ message ControllerStartResponse {
 }
 
 message ControllerPlatformRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	string sandboxer = 10;
 }
 
@@ -139,7 +139,7 @@ message ControllerPlatformResponse {
 }
 
 message ControllerStopRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	uint32 timeout_secs = 2;
 	string sandboxer = 10;
 }
@@ -147,7 +147,7 @@ message ControllerStopRequest {
 message ControllerStopResponse {}
 
 message ControllerWaitRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	string sandboxer = 10;
 }
 
@@ -157,13 +157,13 @@ message ControllerWaitResponse {
 }
 
 message ControllerStatusRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	bool verbose = 2;
 	string sandboxer = 10;
 }
 
 message ControllerStatusResponse {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	uint32 pid = 2;
 	string state = 3;
 	map<string, string> info = 4;
@@ -178,14 +178,14 @@ message ControllerStatusResponse {
 }
 
 message ControllerShutdownRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	string sandboxer = 10;
 }
 
 message ControllerShutdownResponse {}
 
 message ControllerMetricsRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	string sandboxer = 10;
 }
 
@@ -194,7 +194,7 @@ message ControllerMetricsResponse {
 }
 
 message ControllerUpdateRequest {
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	string sandboxer = 2;
 	containerd.types.Sandbox sandbox = 3;
 	repeated string fields = 4;
diff services/streaming/v1/streaming.proto services/streaming/v1/streaming.proto
--- services/streaming/v1/streaming.proto
+++ services/streaming/v1/streaming.proto
@@ -27,5 +27,5 @@ service Streaming {
 }
 
 message StreamInit {
-	string id = 1;
+	string i_d = 1;
 }
diff services/tasks/v1/tasks.proto services/tasks/v1/tasks.proto
--- services/tasks/v1/tasks.proto
+++ services/tasks/v1/tasks.proto
@@ -69,7 +69,7 @@ service Tasks {
 }
 
 message CreateTaskRequest {
-	string container_id = 1;
+	string container_i_d = 1;
 
 	// RootFS provides the pre-chroot mounts to perform in the shim before
 	// executing the container task.
@@ -92,13 +92,13 @@ message CreateTaskRequest {
 }
 
 message CreateTaskResponse {
-	string container_id = 1;
+	string container_i_d = 1;
 	uint32 pid = 2;
 }
 
 message StartRequest {
-	string container_id = 1;
-	string exec_id = 2;
+	string container_i_d = 1;
+	string exec_i_d = 2;
 }
 
 message StartResponse {
@@ -106,24 +106,24 @@ message StartResponse {
 }
 
 message DeleteTaskRequest {
-	string container_id = 1;
+	string container_i_d = 1;
 }
 
 message DeleteResponse {
-	string id = 1;
+	string i_d = 1;
 	uint32 pid = 2;
 	uint32 exit_status = 3;
 	google.protobuf.Timestamp exited_at = 4;
 }
 
 message DeleteProcessRequest {
-	string container_id = 1;
-	string exec_id = 2;
+	string container_i_d = 1;
+	string exec_i_d = 2;
 }
 
 message GetRequest {
-	string container_id = 1;
-	string exec_id = 2;
+	string container_i_d = 1;
+	string exec_i_d = 2;
 }
 
 message GetResponse {
@@ -139,14 +139,14 @@ message ListTasksResponse {
 }
 
 message KillRequest {
-	string container_id = 1;
-	string exec_id = 2;
+	string container_i_d = 1;
+	string exec_i_d = 2;
 	uint32 signal = 3;
 	bool all = 4;
 }
 
 message ExecProcessRequest {
-	string container_id = 1;
+	string container_i_d = 1;
 	string stdin = 2;
 	string stdout = 3;
 	string stderr = 4;
@@ -155,36 +155,36 @@ message ExecProcessRequest {
 	//
 	// For runc, this is a process spec, for example.
 	google.protobuf.Any spec = 6;
-	// id of the exec process
-	string exec_id = 7;
+	// i_d of the exec process
+	string exec_i_d = 7;
 }
 
 message ExecProcessResponse {
 }
 
 message ResizePtyRequest {
-	string container_id = 1;
-	string exec_id = 2;
+	string container_i_d = 1;
+	string exec_i_d = 2;
 	uint32 width = 3;
 	uint32 height = 4;
 }
 
 message CloseIORequest {
-	string container_id = 1;
-	string exec_id = 2;
+	string container_i_d = 1;
+	string exec_i_d = 2;
 	bool stdin = 3;
 }
 
 message PauseTaskRequest {
-	string container_id = 1;
+	string container_i_d = 1;
 }
 
 message ResumeTaskRequest {
-	string container_id = 1;
+	string container_i_d = 1;
 }
 
 message ListPidsRequest {
-	string container_id = 1;
+	string container_i_d = 1;
 }
 
 message ListPidsResponse {
@@ -193,7 +193,7 @@ message ListPidsResponse {
 }
 
 message CheckpointTaskRequest {
-	string container_id = 1;
+	string container_i_d = 1;
 	string parent_checkpoint = 2;
 	google.protobuf.Any options = 3;
 }
@@ -203,7 +203,7 @@ message CheckpointTaskResponse {
 }
 
 message UpdateTaskRequest {
-	string container_id = 1;
+	string container_i_d = 1;
 	google.protobuf.Any resources = 2;
 	map<string, string> annotations = 3;
 }
@@ -217,8 +217,8 @@ message MetricsResponse {
 }
 
 message WaitRequest {
-	string container_id = 1;
-	string exec_id = 2;
+	string container_i_d = 1;
+	string exec_i_d = 2;
 }
 
 message WaitResponse {
diff types/metrics.proto types/metrics.proto
--- types/metrics.proto
+++ types/metrics.proto
@@ -25,6 +25,6 @@ option go_package = "github.com/containerd/containerd/api/types;types";
 
 message Metric {
 	google.protobuf.Timestamp timestamp = 1;
-	string id = 2;
+	string i_d = 2;
 	google.protobuf.Any data = 3;
 }
diff types/platform.proto types/platform.proto
--- types/platform.proto
+++ types/platform.proto
@@ -23,7 +23,7 @@ option go_package = "github.com/containerd/containerd/api/types;types";
 // Platform follows the structure of the OCI platform specification, from
 // descriptors.
 message Platform {
-	string os = 1;
+	string o_s = 1;
 	string architecture = 2;
 	string variant = 3;
-	string os_version = 4;
+	string o_s_version = 4;
 }
diff types/runc/options/oci.proto types/runc/options/oci.proto
--- types/runc/options/oci.proto
+++ types/runc/options/oci.proto
@@ -59,5 +59,5 @@ message CheckpointOptions {
 
 message ProcessDetails {
 	// exec process id if the process is managed by a shim
-	string exec_id = 1;
+	string exec_i_d = 1;
 }
diff types/sandbox.proto types/sandbox.proto
--- types/sandbox.proto
+++ types/sandbox.proto
@@ -27,7 +27,7 @@ option go_package = "github.com/containerd/containerd/api/types;types";
 // work with a particular instance.
 message Sandbox {
 	// SandboxID is a unique instance identifier within namespace
-	string sandbox_id = 1;
+	string sandbox_i_d = 1;
 	message Runtime {
 		// Name is the name of the runtime.
 		string name = 1;
diff types/task/task.proto types/task/task.proto
--- types/task/task.proto
+++ types/task/task.proto
@@ -33,8 +33,8 @@ enum Status {
 }
 
 message Process {
-	string container_id = 1;
-	string id = 2;
+	string container_i_d = 1;
+	string i_d = 2;
 	uint32 pid = 3;
 	Status status = 4;
 	string stdin = 5;
