diff --git a/prometheus/BUILD.bazel b/prometheus/BUILD.bazel
index 32091a5..bb3c51d 100644
--- a/prometheus/BUILD.bazel
+++ b/prometheus/BUILD.bazel
@@ -39,7 +39,7 @@ go_library(
         "//prometheus/internal",
         "@com_github_beorn7_perks//quantile",
         "@com_github_cespare_xxhash_v2//:xxhash",
-        "@com_github_prometheus_client_model//go",
+        "@@//proto:prometheus_client_go_proto",
         "@com_github_prometheus_common//expfmt",
         "@com_github_prometheus_common//model",
         "@org_golang_google_protobuf//proto",
@@ -134,7 +134,7 @@ go_test(
     deps = [
         "//prometheus/internal",
         "//prometheus/promhttp",
-        "@com_github_prometheus_client_model//go",
+        "@@//proto:prometheus_client_go_proto",
         "@com_github_prometheus_common//expfmt",
         "@org_golang_google_protobuf//encoding/protojson",
         "@org_golang_google_protobuf//proto",
diff --git a/prometheus/graphite/BUILD.bazel b/prometheus/graphite/BUILD.bazel
index 17685c8..28c7ccf 100644
--- a/prometheus/graphite/BUILD.bazel
+++ b/prometheus/graphite/BUILD.bazel
@@ -7,7 +7,7 @@ go_library(
     visibility = ["//visibility:public"],
     deps = [
         "//prometheus",
-        "@com_github_prometheus_client_model//go",
+        "@@//proto:prometheus_client_go_proto",
         "@com_github_prometheus_common//expfmt",
         "@com_github_prometheus_common//model",
     ],
diff --git a/prometheus/internal/BUILD.bazel b/prometheus/internal/BUILD.bazel
index 0d0d51e..86b73d5 100644
--- a/prometheus/internal/BUILD.bazel
+++ b/prometheus/internal/BUILD.bazel
@@ -12,7 +12,7 @@ go_library(
     importpath = "github.com/prometheus/client_golang/prometheus/internal",
     visibility = ["//prometheus:__subpackages__"],
     deps = [
-        "@com_github_prometheus_client_model//go",
+        "@@//proto:prometheus_client_go_proto",
         "@com_github_prometheus_common//model",
     ],
 )
diff --git a/prometheus/promhttp/BUILD.bazel b/prometheus/promhttp/BUILD.bazel
index 50fcf73..9c4f0d3 100644
--- a/prometheus/promhttp/BUILD.bazel
+++ b/prometheus/promhttp/BUILD.bazel
@@ -13,7 +13,7 @@ go_library(
     visibility = ["//visibility:public"],
     deps = [
         "//prometheus",
-        "@com_github_prometheus_client_model//go",
+        "@@//proto:prometheus_client_go_proto",
         "@com_github_prometheus_common//expfmt",
     ],
 )
@@ -36,7 +36,7 @@ go_test(
     deps = [
         "//prometheus",
         "//prometheus/testutil",
-        "@com_github_prometheus_client_model//go",
+        "@@//proto:prometheus_client_go_proto",
         "@org_golang_google_protobuf//proto",
     ],
 )
diff --git a/prometheus/testutil/BUILD.bazel b/prometheus/testutil/BUILD.bazel
index 9895c10..be043bc 100644
--- a/prometheus/testutil/BUILD.bazel
+++ b/prometheus/testutil/BUILD.bazel
@@ -13,7 +13,7 @@ go_library(
         "//prometheus/internal",
         "//prometheus/testutil/promlint",
         "@com_github_davecgh_go_spew//spew",
-        "@com_github_prometheus_client_model//go",
+        "@@//proto:prometheus_client_go_proto",
         "@com_github_prometheus_common//expfmt",
         "@org_golang_google_protobuf//proto",
     ],
diff --git a/prometheus/testutil/promlint/BUILD.bazel b/prometheus/testutil/promlint/BUILD.bazel
index 61bad92..b9c01c9 100644
--- a/prometheus/testutil/promlint/BUILD.bazel
+++ b/prometheus/testutil/promlint/BUILD.bazel
@@ -11,7 +11,7 @@ go_library(
     visibility = ["//visibility:public"],
     deps = [
         "//prometheus/testutil/promlint/validations",
-        "@com_github_prometheus_client_model//go",
+        "@@//proto:prometheus_client_go_proto",
         "@com_github_prometheus_common//expfmt",
     ],
 )
@@ -27,6 +27,6 @@ go_test(
     srcs = ["promlint_test.go"],
     deps = [
         ":promlint",
-        "@com_github_prometheus_client_model//go",
+        "@@//proto:prometheus_client_go_proto",
     ],
 )
diff --git a/prometheus/testutil/promlint/validations/BUILD.bazel b/prometheus/testutil/promlint/validations/BUILD.bazel
index 5c135ce..9604a0f 100644
--- a/prometheus/testutil/promlint/validations/BUILD.bazel
+++ b/prometheus/testutil/promlint/validations/BUILD.bazel
@@ -11,7 +11,7 @@ go_library(
     ],
     importpath = "github.com/prometheus/client_golang/prometheus/testutil/promlint/validations",
     visibility = ["//visibility:public"],
-    deps = ["@com_github_prometheus_client_model//go"],
+    deps = ["@@//proto:prometheus_client_go_proto"],
 )
 
 alias(
