From ff231b39ae8c2c4162b167a3148080f25ad3b2d0 Mon Sep 17 00:00:00 2001
From: Zoey Greer <zoey@buildbuddy.io>
Date: Mon, 4 Dec 2023 11:06:58 -0500
Subject: [PATCH] did it


diff --git a/BUILD b/BUILD
new file mode 100644
index 0000000..e752152
--- /dev/null
+++ b/BUILD
@@ -0,0 +1,117 @@
+package(default_visibility = ["//visibility:public"])
+
+filegroup(
+    name = "x86_64-full",
+    srcs = [
+        "//:x86_64-bin",
+        "//:libexec/gcc/x86_64-linux-cosmo/11.2.0",
+        "//:include",
+        "//:x86_64-lib",
+    ],
+)
+
+# include
+
+filegroup(
+    name = "include",
+    srcs = glob(["include/**/*"]),
+)
+
+# bin
+
+filegroup(
+    name = "x86_64-bin",
+    srcs = [
+        "x86_64-ar",
+        "x86_64-gcc",
+        "x86_64-g++",
+        "x86_64-strip",
+        "x86_64-objcopy",
+        "x86_64-objdump",
+        "x86_64-addr2line",
+        "x86_64-elfedit",
+        "x86_64-nm",
+        "x86_64-ranlib",
+        "x86_64-readelf",
+        "x86_64-size",
+        "x86_64-c++filt",
+    ],
+)
+
+filegroup(name = "x86_64-ar", srcs = ["bin/x86_64-linux-cosmo-ar"])
+filegroup(name = "x86_64-gcc", srcs = ["bin/x86_64-linux-cosmo-gcc"])
+filegroup(name = "x86_64-g++", srcs = ["bin/x86_64-linux-cosmo-g++"])
+filegroup(name = "x86_64-strip", srcs = ["bin/x86_64-linux-cosmo-strip"])
+filegroup(name = "x86_64-objcopy", srcs = ["bin/x86_64-linux-cosmo-objcopy"])
+filegroup(name = "x86_64-objdump", srcs = ["bin/x86_64-linux-cosmo-objdump"])
+filegroup(name = "x86_64-addr2line", srcs = ["bin/x86_64-linux-cosmo-addr2line"])
+filegroup(name = "x86_64-elfedit", srcs = ["bin/x86_64-linux-cosmo-elfedit"])
+filegroup(name = "x86_64-nm", srcs = ["bin/x86_64-linux-cosmo-nm"])
+filegroup(name = "x86_64-ranlib", srcs = ["bin/x86_64-linux-cosmo-ranlib"])
+filegroup(name = "x86_64-readelf", srcs = ["bin/x86_64-linux-cosmo-readelf"])
+filegroup(name = "x86_64-size", srcs = ["bin/x86_64-linux-cosmo-size"])
+filegroup(name = "x86_64-c++filt", srcs = ["bin/x86_64-linux-cosmo-c++filt"])
+
+alias(name = "x86_64-c++", actual = ":x86_64-g++")
+alias(name = "x86_64-cc", actual = ":x86_64-gcc")
+alias(name = "x86_64-cpp", actual = ":x86_64-gcc")
+
+# libexec
+
+filegroup(
+    name = "libexec/gcc/x86_64-linux-cosmo/11.2.0",
+    srcs = [
+        "x86_64-as",
+        "x86_64-cc1",
+        "x86_64-cc1plus",
+        "x86_64-collect2",
+        "x86_64-ld.bfd",
+    ],
+)
+
+filegroup(name = "x86_64-as", srcs = ["libexec/gcc/x86_64-linux-cosmo/11.2.0/as"])
+filegroup(name = "x86_64-cc1", srcs = ["libexec/gcc/x86_64-linux-cosmo/11.2.0/cc1"])
+filegroup(name = "x86_64-cc1plus", srcs = ["libexec/gcc/x86_64-linux-cosmo/11.2.0/cc1plus"])
+filegroup(name = "x86_64-collect2", srcs = ["libexec/gcc/x86_64-linux-cosmo/11.2.0/collect2"])
+filegroup(name = "x86_64-ld.bfd", srcs = ["libexec/gcc/x86_64-linux-cosmo/11.2.0/ld.bfd"])
+
+alias(name = "x86_64-ld", actual = ":x86_64-ld.bfd")
+
+# lib
+
+filegroup(
+    name = "x86_64-lib",
+    srcs = [
+        ":x86_64-ape-no-modify-self.o",
+        ":x86_64-ape.lds",
+        ":x86_64-ape.o",
+        ":x86_64-crt.o",
+        ":x86_64-libc.a",
+        ":x86_64-libcosmo.a",
+        ":x86_64-libcxx.a",
+        ":x86_64-libdl.a",
+        ":x86_64-libgcc_s.a",
+        ":x86_64-libm.a",
+        ":x86_64-libpthread.a",
+        ":x86_64-libresolv.a",
+        ":x86_64-librt.a",
+        ":x86_64-libstdc++.a",
+        ":x86_64-libz.a",
+    ],
+)
+
+filegroup(name = "x86_64-ape-no-modify-self.o", srcs = ["x86_64-linux-cosmo/lib/ape-no-modify-self.o"])
+filegroup(name = "x86_64-ape.lds", srcs = ["x86_64-linux-cosmo/lib/ape.lds"])
+filegroup(name = "x86_64-ape.o", srcs = ["x86_64-linux-cosmo/lib/ape.o"])
+filegroup(name = "x86_64-crt.o", srcs = ["x86_64-linux-cosmo/lib/crt.o"])
+filegroup(name = "x86_64-libc.a", srcs = ["x86_64-linux-cosmo/lib/libc.a"])
+filegroup(name = "x86_64-libcosmo.a", srcs = ["x86_64-linux-cosmo/lib/libcosmo.a"])
+filegroup(name = "x86_64-libcxx.a", srcs = ["x86_64-linux-cosmo/lib/libcxx.a"])
+filegroup(name = "x86_64-libdl.a", srcs = ["x86_64-linux-cosmo/lib/libdl.a"])
+filegroup(name = "x86_64-libgcc_s.a", srcs = ["x86_64-linux-cosmo/lib/libgcc_s.a"])
+filegroup(name = "x86_64-libm.a", srcs = ["x86_64-linux-cosmo/lib/libm.a"])
+filegroup(name = "x86_64-libpthread.a", srcs = ["x86_64-linux-cosmo/lib/libpthread.a"])
+filegroup(name = "x86_64-libresolv.a", srcs = ["x86_64-linux-cosmo/lib/libresolv.a"])
+filegroup(name = "x86_64-librt.a", srcs = ["x86_64-linux-cosmo/lib/librt.a"])
+filegroup(name = "x86_64-libstdc++.a", srcs = ["x86_64-linux-cosmo/lib/libstdc++.a"])
+filegroup(name = "x86_64-libz.a", srcs = ["x86_64-linux-cosmo/lib/libz.a"])
diff --git a/WORKSPACE b/WORKSPACE
new file mode 100644
index 0000000..0188bcc
--- /dev/null
+++ b/WORKSPACE
@@ -0,0 +1,3 @@
+workspace(
+    name = "cosmocc",
+)
-- 
2.39.2

