diff --git a/proto/cel/expr/BUILD.bazel b/proto/cel/expr/BUILD.bazel
index d80de43..29db706 100644
--- proto/cel/expr/BUILD.bazel
+++ proto/cel/expr/BUILD.bazel
@@ -72,8 +72,8 @@ go_proto_library(
     ],
     visibility = ["//visibility:public"],
     deps = [
-        ":syntax_go_proto",
-        ":value_go_proto",
         "@org_golang_google_genproto_googleapis_rpc//status",
+        "@org_golang_google_protobuf//reflect/protoreflect",
+        "@org_golang_google_protobuf//runtime/protoimpl",
     ],
 )
