diff --git a/client/hijack.go b/client/hijack.go
index 6bdacab10a..aaf3e9ab09 100644
--- a/client/hijack.go
+++ b/client/hijack.go
@@ -9,6 +9,7 @@ import (
 	"net/http"
 	"net/http/httputil"
 	"net/url"
+	"strings"
 	"time"
 
 	"github.com/docker/docker/api/types"
@@ -64,7 +65,11 @@ func fallbackDial(proto, addr string, tlsConfig *tls.Config) (net.Conn, error) {
 }
 
 func (cli *Client) setupHijackConn(ctx context.Context, req *http.Request, proto string) (net.Conn, string, error) {
-	req.Host = cli.addr
+	if strings.HasSuffix(cli.addr, ".sock") {
+		req.Host = "localhost"
+	} else {
+		req.Host = cli.addr
+	}
 	req.Header.Set("Connection", "Upgrade")
 	req.Header.Set("Upgrade", proto)
 
