diff --git a/fastcdc.go b/fastcdc.go
index 85f8ffd..24e7acc 100644
--- a/fastcdc.go
+++ b/fastcdc.go
@@ -106,8 +106,10 @@ func NewChunker(rd io.Reader, opts Options) (*Chunker, error) {
 		return nil, err
 	}
 
-	for i := 0; i < len(table); i++ {
-		table[i] = table[i] ^ opts.Seed
+	if opts.Seed != 0 {
+		for i := 0; i < len(table); i++ {
+			table[i] = table[i] ^ opts.Seed
+		}
 	}
 
 	normalization := opts.Normalization
