diff -ruN a/cmd/gazelle/BUILD.bazel b/cmd/gazelle/BUILD.bazel
--- a/cmd/gazelle/BUILD.bazel	2022-10-25 15:58:36.000000000 -0700
+++ b/cmd/gazelle/BUILD.bazel	2023-01-27 15:06:05.000000000 -0800
@@ -18,6 +18,7 @@
         "metaresolver.go",
         "print.go",
         "update-repos.go",
+        "langs.go",
     ],
     importpath = "github.com/bazelbuild/bazel-gazelle/cmd/gazelle",
     tags = ["manual"],
diff -ruN a/cmd/gazelle/gazelle.go b/cmd/gazelle/gazelle.go
--- a/cmd/gazelle/gazelle.go	2022-10-25 15:58:36.000000000 -0700
+++ b/cmd/gazelle/gazelle.go	2023-01-27 14:26:20.000000000 -0800
@@ -56,6 +56,10 @@
 }

 func main() {
+	Main()
+}
+
+func Main() {
 	log.SetPrefix("gazelle: ")
 	log.SetFlags(0) // don't print timestamps

	var wd string