diff --git a/WORKSPACE b/WORKSPACE
index 009de188b..fe3622e26 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -87,17 +87,6 @@ ruby_bundle(
     gemfile_lock = "//kythe/web/site:Gemfile.lock",
 )
 
-http_archive(
-    name = "aspect_bazel_lib",
-    sha256 = "d488d8ecca98a4042442a4ae5f1ab0b614f896c0ebf6e3eafff363bcc51c6e62",
-    strip_prefix = "bazel-lib-1.33.0",
-    url = "https://github.com/aspect-build/bazel-lib/releases/download/v1.33.0/bazel-lib-v1.33.0.tar.gz",
-)
-
-load("@aspect_bazel_lib//lib:repositories.bzl", "aspect_bazel_lib_dependencies")
-
-aspect_bazel_lib_dependencies()
-
 # clang-tidy aspect wrapper
 load(
     "@bazel_tools//tools/build_defs/repo:git.bzl",
diff --git a/kythe/go/services/cli/BUILD b/kythe/go/services/cli/BUILD
index 17bc7623d..a8536dd9b 100644
--- a/kythe/go/services/cli/BUILD
+++ b/kythe/go/services/cli/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "cli",
diff --git a/kythe/go/services/explore/BUILD b/kythe/go/services/explore/BUILD
index a04f92ee5..4578601c9 100644
--- a/kythe/go/services/explore/BUILD
+++ b/kythe/go/services/explore/BUILD
@@ -1,7 +1,7 @@
 load("//tools:build_rules/shims.bzl", "go_library")
 
 package(
-    default_visibility = ["//kythe:default_visibility"],
+    default_visibility = ["//visibility:public"],
     licenses = ["notice"],
 )
 
diff --git a/kythe/go/services/filetree/BUILD b/kythe/go/services/filetree/BUILD
index de08c571f..cc22169a7 100644
--- a/kythe/go/services/filetree/BUILD
+++ b/kythe/go/services/filetree/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "filetree",
diff --git a/kythe/go/services/graph/BUILD b/kythe/go/services/graph/BUILD
index 527102ea4..80b030af2 100644
--- a/kythe/go/services/graph/BUILD
+++ b/kythe/go/services/graph/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "graph",
diff --git a/kythe/go/services/graphstore/BUILD b/kythe/go/services/graphstore/BUILD
index 9e3ecb846..6d2cc1088 100644
--- a/kythe/go/services/graphstore/BUILD
+++ b/kythe/go/services/graphstore/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "graphstore",
diff --git a/kythe/go/services/link/BUILD b/kythe/go/services/link/BUILD
index 8fa2c4842..c4b7017d1 100644
--- a/kythe/go/services/link/BUILD
+++ b/kythe/go/services/link/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "link",
diff --git a/kythe/go/services/web/BUILD b/kythe/go/services/web/BUILD
index aefd111c7..536d6dd82 100644
--- a/kythe/go/services/web/BUILD
+++ b/kythe/go/services/web/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "web",
diff --git a/kythe/go/services/xrefs/BUILD b/kythe/go/services/xrefs/BUILD
index 3ad95120d..873acc05d 100644
--- a/kythe/go/services/xrefs/BUILD
+++ b/kythe/go/services/xrefs/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "xrefs",
diff --git a/kythe/go/serving/api/BUILD b/kythe/go/serving/api/BUILD
index c3ef56ddf..975fb00fc 100644
--- a/kythe/go/serving/api/BUILD
+++ b/kythe/go/serving/api/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "api",
diff --git a/kythe/go/serving/explore/BUILD b/kythe/go/serving/explore/BUILD
index bcdb82c73..a35fdf686 100644
--- a/kythe/go/serving/explore/BUILD
+++ b/kythe/go/serving/explore/BUILD
@@ -1,7 +1,7 @@
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
 package(
-    default_visibility = ["//kythe:default_visibility"],
+    default_visibility = ["//visibility:public"],
     licenses = ["notice"],
 )
 
diff --git a/kythe/go/serving/filetree/BUILD b/kythe/go/serving/filetree/BUILD
index 4e3141c3b..3c527d529 100644
--- a/kythe/go/serving/filetree/BUILD
+++ b/kythe/go/serving/filetree/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "filetree",
diff --git a/kythe/go/serving/graph/BUILD b/kythe/go/serving/graph/BUILD
index 3227129af..78768748f 100644
--- a/kythe/go/serving/graph/BUILD
+++ b/kythe/go/serving/graph/BUILD
@@ -1,7 +1,7 @@
 # gazelle:go_test file
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "graph",
diff --git a/kythe/go/serving/identifiers/BUILD b/kythe/go/serving/identifiers/BUILD
index 6286df42a..17c9d9904 100644
--- a/kythe/go/serving/identifiers/BUILD
+++ b/kythe/go/serving/identifiers/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "identifiers",
diff --git a/kythe/go/serving/pipeline/BUILD b/kythe/go/serving/pipeline/BUILD
index dd9256ac6..1cbb67730 100644
--- a/kythe/go/serving/pipeline/BUILD
+++ b/kythe/go/serving/pipeline/BUILD
@@ -1,7 +1,7 @@
 # gazelle:go_test file
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "pipeline",
diff --git a/kythe/go/serving/tools/BUILD b/kythe/go/serving/tools/BUILD
index 0fdca657e..8d498f521 100644
--- a/kythe/go/serving/tools/BUILD
+++ b/kythe/go/serving/tools/BUILD
@@ -1,4 +1,4 @@
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 filegroup(
     name = "http_server",
diff --git a/kythe/go/serving/xrefs/BUILD b/kythe/go/serving/xrefs/BUILD
index 8dde37862..767a0f526 100644
--- a/kythe/go/serving/xrefs/BUILD
+++ b/kythe/go/serving/xrefs/BUILD
@@ -1,7 +1,7 @@
 # gazelle:go_test file
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "xrefs",
diff --git a/kythe/go/storage/entryset/BUILD b/kythe/go/storage/entryset/BUILD
index 315f044de..0b7264cd2 100644
--- a/kythe/go/storage/entryset/BUILD
+++ b/kythe/go/storage/entryset/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "entryset",
diff --git a/kythe/go/storage/gsutil/BUILD b/kythe/go/storage/gsutil/BUILD
index 60551629b..5d2ec6541 100644
--- a/kythe/go/storage/gsutil/BUILD
+++ b/kythe/go/storage/gsutil/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "gsutil",
diff --git a/kythe/go/storage/inmemory/BUILD b/kythe/go/storage/inmemory/BUILD
index 3e19ff66c..9bd292f50 100644
--- a/kythe/go/storage/inmemory/BUILD
+++ b/kythe/go/storage/inmemory/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "inmemory",
diff --git a/kythe/go/storage/keyvalue/BUILD b/kythe/go/storage/keyvalue/BUILD
index 7717e6de1..84d82587c 100644
--- a/kythe/go/storage/keyvalue/BUILD
+++ b/kythe/go/storage/keyvalue/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "keyvalue",
diff --git a/kythe/go/storage/kvutil/BUILD b/kythe/go/storage/kvutil/BUILD
index 6a813fb0a..29227dab7 100644
--- a/kythe/go/storage/kvutil/BUILD
+++ b/kythe/go/storage/kvutil/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "kvutil",
diff --git a/kythe/go/storage/leveldb/BUILD b/kythe/go/storage/leveldb/BUILD
index b96088fb4..85c576f11 100644
--- a/kythe/go/storage/leveldb/BUILD
+++ b/kythe/go/storage/leveldb/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "leveldb",
diff --git a/kythe/go/storage/pebble/BUILD b/kythe/go/storage/pebble/BUILD
index 299ae3a53..5717f2d6c 100644
--- a/kythe/go/storage/pebble/BUILD
+++ b/kythe/go/storage/pebble/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "pebble",
diff --git a/kythe/go/storage/stream/BUILD b/kythe/go/storage/stream/BUILD
index fd7bcea9c..1506f6e0c 100644
--- a/kythe/go/storage/stream/BUILD
+++ b/kythe/go/storage/stream/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "stream",
diff --git a/kythe/go/storage/table/BUILD b/kythe/go/storage/table/BUILD
index 4b3ecd2fd..1c42406a2 100644
--- a/kythe/go/storage/table/BUILD
+++ b/kythe/go/storage/table/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_library")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_library(
     name = "table",
diff --git a/kythe/go/storage/tools/BUILD b/kythe/go/storage/tools/BUILD
index db7250a1e..afa648bb7 100644
--- a/kythe/go/storage/tools/BUILD
+++ b/kythe/go/storage/tools/BUILD
@@ -1,4 +1,4 @@
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 filegroup(
     name = "write_entries",
diff --git a/kythe/go/storage/tools/directory_indexer/BUILD b/kythe/go/storage/tools/directory_indexer/BUILD
index 882156aa3..04ac231fe 100644
--- a/kythe/go/storage/tools/directory_indexer/BUILD
+++ b/kythe/go/storage/tools/directory_indexer/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_binary")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_binary(
     name = "directory_indexer",
diff --git a/kythe/go/storage/tools/read_entries/BUILD b/kythe/go/storage/tools/read_entries/BUILD
index 857198193..6053b9497 100644
--- a/kythe/go/storage/tools/read_entries/BUILD
+++ b/kythe/go/storage/tools/read_entries/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_binary")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_binary(
     name = "read_entries",
diff --git a/kythe/go/storage/tools/triples/BUILD b/kythe/go/storage/tools/triples/BUILD
index 17858ad65..7d885b8dc 100644
--- a/kythe/go/storage/tools/triples/BUILD
+++ b/kythe/go/storage/tools/triples/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_binary")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_binary(
     name = "triples",
diff --git a/kythe/go/storage/tools/write_entries/BUILD b/kythe/go/storage/tools/write_entries/BUILD
index 6d64d8925..c6409f7ad 100644
--- a/kythe/go/storage/tools/write_entries/BUILD
+++ b/kythe/go/storage/tools/write_entries/BUILD
@@ -1,6 +1,6 @@
 load("//tools:build_rules/shims.bzl", "go_binary")
 
-package(default_visibility = ["//kythe:default_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 go_binary(
     name = "write_entries",
diff --git a/kythe/go/util/schema/BUILD b/kythe/go/util/schema/BUILD
index e2c81cb36..9629ae56f 100644
--- a/kythe/go/util/schema/BUILD
+++ b/kythe/go/util/schema/BUILD
@@ -1,7 +1,6 @@
 # The checked-in generated files confuse gazelle.
 # gazelle:ignore
 load("//tools:build_rules/shims.bzl", "go_library", "go_test")
-load("@aspect_bazel_lib//lib:write_source_files.bzl", "write_source_file")
 
 package(default_visibility = ["//kythe:default_visibility"])
 
@@ -30,12 +29,6 @@ genrule(
     visibility = ["//visibility:private"],
 )
 
-write_source_file(
-    name = "schema_index_sync",
-    in_file = ":schema_index",
-    out_file = "indexdata.go",
-)
-
 go_test(
     name = "schema_test",
     size = "small",
diff --git a/kythe/proto/BUILD b/kythe/proto/BUILD
index b76bccaea..7cd5b6290 100644
--- a/kythe/proto/BUILD
+++ b/kythe/proto/BUILD
@@ -1,11 +1,9 @@
 load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
 load("@rules_proto//proto:defs.bzl", "proto_library")
-load("@rules_java//java:defs.bzl", "java_proto_library")
 load("//:visibility.bzl", "PUBLIC_PROTO_VISIBILITY")
-load(":genproto.bzl", "update_generated_protos")
 load(":go.bzl", "go_proto_library")
 
-package(default_visibility = ["//kythe:proto_visibility"])
+package(default_visibility = ["//visibility:public"])
 
 filegroup(
     name = "public",
@@ -47,12 +45,6 @@ go_proto_library(
     deps = [":storage_go_proto"],
 )
 
-java_proto_library(
-    name = "schema_java_proto",
-    visibility = [PUBLIC_PROTO_VISIBILITY],
-    deps = [":schema_proto"],
-)
-
 cc_proto_library(
     name = "schema_cc_proto",
     visibility = [PUBLIC_PROTO_VISIBILITY],
@@ -79,18 +71,12 @@ go_proto_library(
     proto = ":common_proto",
 )
 
-java_proto_library(
-    name = "common_java_proto",
-    visibility = [PUBLIC_PROTO_VISIBILITY],
-    deps = [":common_proto"],
-)
-
 # Public Kythe GraphStore analyzer output format
 proto_library(
     name = "storage_proto",
     srcs = ["storage.proto"],
     visibility = [
-        "//kythe:proto_visibility",
+        "//visibility:public",
     ],
 )
 
@@ -106,18 +92,12 @@ go_proto_library(
     proto = ":storage_proto",
 )
 
-java_proto_library(
-    name = "storage_java_proto",
-    visibility = [PUBLIC_PROTO_VISIBILITY],
-    deps = [":storage_proto"],
-)
-
 # Public Kythe GraphStore service API
 proto_library(
     name = "storage_service_proto",
     srcs = ["storage_service.proto"],
     visibility = [
-        "//kythe:proto_visibility",
+        "//visibility:public",
     ],
     deps = [":storage_proto"],
 )
@@ -129,24 +109,19 @@ cc_proto_library(
 
 go_proto_library(
     name = "storage_service_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_proto", "@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "kythe.io/kythe/proto/storage_service_go_proto",
     proto = ":storage_service_proto",
     visibility = [PUBLIC_PROTO_VISIBILITY],
     deps = [":storage_go_proto"],
 )
 
-java_proto_library(
-    name = "storage_service_java_proto",
-    deps = [":storage_service_proto"],
-)
-
 # Public Kythe analyzer
 proto_library(
     name = "analysis_proto",
     srcs = ["analysis.proto"],
     visibility = [
-        "//kythe:proto_visibility",
+        "//visibility:public",
     ],
     deps = [
         ":storage_proto",
@@ -171,18 +146,12 @@ go_proto_library(
     ],
 )
 
-java_proto_library(
-    name = "analysis_java_proto",
-    visibility = [PUBLIC_PROTO_VISIBILITY],
-    deps = [":analysis_proto"],
-)
-
 # Public Kythe file data service APIs
 proto_library(
     name = "analysis_service_proto",
     srcs = ["analysis_service.proto"],
     visibility = [
-        "//kythe:proto_visibility",
+        "//visibility:public",
     ],
     deps = [":analysis_proto"],
 )
@@ -194,17 +163,12 @@ cc_proto_library(
 
 go_proto_library(
     name = "analysis_service_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_proto", "@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "kythe.io/kythe/proto/analysis_service_go_proto",
     proto = ":analysis_service_proto",
     deps = [":analysis_go_proto"],
 )
 
-java_proto_library(
-    name = "analysis_service_java_proto",
-    deps = [":analysis_service_proto"],
-)
-
 # Public Kythe status service API
 proto_library(
     name = "status_service_proto",
@@ -225,11 +189,6 @@ go_proto_library(
     deps = [":common_go_proto"],
 )
 
-java_proto_library(
-    name = "status_service_java_proto",
-    deps = [":status_service_proto"],
-)
-
 # Context-dependent version information, for claiming.
 proto_library(
     name = "filecontext_proto",
@@ -247,11 +206,6 @@ go_proto_library(
     proto = ":filecontext_proto",
 )
 
-java_proto_library(
-    name = "filecontext_java_proto",
-    deps = [":filecontext_proto"],
-)
-
 # Build tool details for compilation units
 proto_library(
     name = "buildinfo_proto",
@@ -270,11 +224,6 @@ go_proto_library(
     proto = ":buildinfo_proto",
 )
 
-java_proto_library(
-    name = "buildinfo_java_proto",
-    deps = [":buildinfo_proto"],
-)
-
 # C++-specific protocol buffer definitions
 proto_library(
     name = "cxx_proto",
@@ -292,11 +241,6 @@ go_proto_library(
     proto = ":cxx_proto",
 )
 
-java_proto_library(
-    name = "cxx_java_proto",
-    deps = [":cxx_proto"],
-)
-
 # Go-specific protocol buffer definitions
 proto_library(
     name = "go_proto",
@@ -314,11 +258,6 @@ go_proto_library(
     proto = ":go_proto",
 )
 
-java_proto_library(
-    name = "go_java_proto",
-    deps = [":go_proto"],
-)
-
 # Java-specific protocol buffer definitions
 proto_library(
     name = "java_proto",
@@ -338,11 +277,6 @@ go_proto_library(
     deps = [":storage_go_proto"],
 )
 
-java_proto_library(
-    name = "java_java_proto",
-    deps = [":java_proto"],
-)
-
 # Kythe claiming message definitions
 proto_library(
     name = "claim_proto",
@@ -362,11 +296,6 @@ go_proto_library(
     deps = [":storage_go_proto"],
 )
 
-java_proto_library(
-    name = "claim_java_proto",
-    deps = [":claim_proto"],
-)
-
 # Public Kythe filetree service API
 proto_library(
     name = "filetree_proto",
@@ -380,32 +309,17 @@ cc_proto_library(
 
 go_proto_library(
     name = "filetree_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_proto", "@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "kythe.io/kythe/proto/filetree_go_proto",
     proto = ":filetree_proto",
 )
 
-java_proto_library(
-    name = "filetree_java_proto",
-    deps = [":filetree_proto"],
-)
-
 # Public Kythe xref service API
 proto_library(
     name = "xref_proto",
     srcs = ["xref.proto"],
     visibility = [
-        # TODO(schroederc): prune once MarkedSource has migrated to common.proto
-        "//kythe/cxx/common:__pkg__",
-        "//kythe/cxx/doc:__pkg__",
-        "//kythe/go/languageserver:__subpackages__",
-        "//kythe/go/platform/tools:__subpackages__",
-        "//kythe/go/services:__subpackages__",
-        "//kythe/go/serving:__subpackages__",
-        "//kythe/go/test:__subpackages__",
-        "//kythe/go/util/markedsource:__pkg__",
-        "//kythe/java/com/google/devtools/kythe/doc:__subpackages__",
-        "//kythe/javatests/com/google/devtools/kythe/doc:__subpackages__",
+        "//visibility:public",
     ],
     deps = [":common_proto"],
 )
@@ -418,17 +332,12 @@ cc_proto_library(
 
 go_proto_library(
     name = "xref_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_proto", "@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "kythe.io/kythe/proto/xref_go_proto",
     proto = ":xref_proto",
     deps = [":common_go_proto"],
 )
 
-java_proto_library(
-    name = "xref_java_proto",
-    deps = [":xref_proto"],
-)
-
 proto_library(
     name = "identifier_proto",
     srcs = ["identifier.proto"],
@@ -441,16 +350,11 @@ cc_proto_library(
 
 go_proto_library(
     name = "identifier_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_proto", "@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "kythe.io/kythe/proto/identifier_go_proto",
     proto = ":identifier_proto",
 )
 
-java_proto_library(
-    name = "identifier_java_proto",
-    deps = [":identifier_proto"],
-)
-
 # Public Kythe graph service API
 proto_library(
     name = "graph_proto",
@@ -460,7 +364,7 @@ proto_library(
 
 go_proto_library(
     name = "graph_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_proto", "@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "kythe.io/kythe/proto/graph_go_proto",
     proto = ":graph_proto",
     deps = [":common_go_proto"],
@@ -471,11 +375,6 @@ cc_proto_library(
     deps = [":graph_proto"],
 )
 
-java_proto_library(
-    name = "graph_java_proto",
-    deps = [":graph_proto"],
-)
-
 # Public Kythe graph exploration API
 proto_library(
     name = "explore_proto",
@@ -489,7 +388,7 @@ proto_library(
 
 go_proto_library(
     name = "explore_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_proto", "@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "kythe.io/kythe/proto/explore_go_proto",
     proto = ":explore_proto",
     deps = [
@@ -504,11 +403,6 @@ cc_proto_library(
     deps = [":explore_proto"],
 )
 
-java_proto_library(
-    name = "explore_java_proto",
-    deps = [":explore_proto"],
-)
-
 # Protocol buffer definitions used to encode serving data
 # WARNING: These should not be exposed to clients.
 proto_library(
@@ -535,12 +429,6 @@ go_proto_library(
     deps = [":common_go_proto"],
 )
 
-java_proto_library(
-    name = "serving_java_proto",
-    visibility = ["//visibility:public"],
-    deps = [":serving_proto"],
-)
-
 proto_library(
     name = "xref_serving_proto",
     srcs = ["xref_serving.proto"],
@@ -598,17 +486,12 @@ cc_proto_library(
 
 go_proto_library(
     name = "link_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_proto", "@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "kythe.io/kythe/proto/link_go_proto",
     proto = ":link_proto",
     deps = [":common_go_proto"],
 )
 
-java_proto_library(
-    name = "link_java_proto",
-    deps = [":link_proto"],
-)
-
 # Protocol buffer definitions internal only to the Kythe libraries and tools.
 # WARNING: These should not be exposed to clients.
 proto_library(
@@ -629,11 +512,6 @@ go_proto_library(
     deps = [":serving_go_proto"],
 )
 
-java_proto_library(
-    name = "internal_java_proto",
-    deps = [":internal_proto"],
-)
-
 # Protocol buffer definitions used to record indexer outputs for storage.
 # WARNING: These should not be exposed to clients.
 proto_library(
@@ -652,11 +530,6 @@ go_proto_library(
     proto = ":entryset_proto",
 )
 
-java_proto_library(
-    name = "entryset_java_proto",
-    deps = [":entryset_proto"],
-)
-
 proto_library(
     name = "extraction_config_proto",
     srcs = ["extraction_config.proto"],
@@ -744,16 +617,6 @@ cc_proto_library(
     deps = [":metadata_proto"],
 )
 
-java_proto_library(
-    name = "metadata_java_proto",
-    deps = [":metadata_proto"],
-)
-
-bzl_library(
-    name = "genproto_bzl",
-    srcs = ["genproto.bzl"],
-)
-
 bzl_library(
     name = "go_bzl",
     srcs = ["go.bzl"],
@@ -817,14 +680,6 @@ go_proto_library(
     visibility = [PUBLIC_PROTO_VISIBILITY],
 )
 
-java_proto_library(
-    name = "generated_message_info_java_proto",
-    visibility = [PUBLIC_PROTO_VISIBILITY],
-    deps = [
-        ":generated_message_info_proto",
-    ],
-)
-
 cc_proto_library(
     name = "generated_message_info_cc_proto",
     visibility = [PUBLIC_PROTO_VISIBILITY],
@@ -850,13 +705,3 @@ go_proto_library(
     proto = ":vnames_config_proto",
     deps = [":storage_go_proto"],
 )
-
-# Uses native.existing_rules() to find rules to update; must come last.
-update_generated_protos(
-    name = "update",
-    additional_update_targets = [
-        # Ensure this updates the protos which live elsewhere.
-        "//kythe/go/util/riegeli:riegeli_test_go_proto_sync",
-        "//third_party/bazel:update",
-    ],
-)
diff --git a/kythe/proto/genproto.bzl b/kythe/proto/genproto.bzl
index ec30a5186..e69de29bb 100644
--- a/kythe/proto/genproto.bzl
+++ b/kythe/proto/genproto.bzl
@@ -1,11 +0,0 @@
-load("@aspect_bazel_lib//lib:write_source_files.bzl", "write_source_files")
-
-def update_generated_protos(name, additional_update_targets = []):
-    write_source_files(
-        name = name,
-        additional_update_targets = additional_update_targets + [
-            key
-            for key, value in native.existing_rules().items()
-            if value["kind"] == "_write_source_file"
-        ],
-    )
diff --git a/kythe/proto/go.bzl b/kythe/proto/go.bzl
index a1035df18..50375e5fc 100644
--- a/kythe/proto/go.bzl
+++ b/kythe/proto/go.bzl
@@ -1,4 +1,3 @@
-load("@aspect_bazel_lib//lib:write_source_files.bzl", "write_source_file")
 load("@io_bazel_rules_go//proto:def.bzl", _go_proto_library = "go_proto_library")
 
 KYTHE_IMPORT_BASE = "kythe.io/kythe/proto"
@@ -59,9 +58,3 @@ def go_proto_library(
         output_group = "go_generated_srcs",
         srcs = [name],
     )
-    write_source_file(
-        name = name + "_sync",
-        in_file = name + "_src",
-        out_file = name + "/" + filename,
-        suggested_update_target = suggested_update_target,
-    )
diff --git a/third_party/bazel/BUILD b/third_party/bazel/BUILD
index bbf24aae3..fb5b7fbd2 100644
--- a/third_party/bazel/BUILD
+++ b/third_party/bazel/BUILD
@@ -5,10 +5,8 @@
 load("@rules_proto//proto:defs.bzl", "proto_library")
 load("@rules_java//java:defs.bzl", "java_proto_library")
 load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
-load("@aspect_bazel_lib//lib:write_source_files.bzl", "write_source_file")
 load("@aspect_bazel_lib//lib:copy_to_directory.bzl", "copy_to_directory")
 load("//kythe/proto:go.bzl", "go_proto_library")
-load("//kythe/proto:genproto.bzl", "update_generated_protos")
 
 package(default_visibility = ["//visibility:public"])
 
@@ -36,14 +34,6 @@ copy_to_directory(
     visibility = ["//visibility:private"],
 )
 
-write_source_file(
-    name = "proto_srcs_sync",
-    in_file = ":proto_srcs",
-    out_file = "src",
-    tags = ["local"],  # Fails spuriously remotely.
-    visibility = ["//visibility:private"],
-)
-
 alias(
     name = "extra_actions_base_proto",
     actual = "//third_party/bazel/src/main/protobuf:extra_actions_base_proto",
@@ -170,5 +160,3 @@ sh_binary(
     name = "get_sdkroot",
     srcs = ["get_sdkroot.sh"],
 )
-
-update_generated_protos(name = "update")
diff --git a/tools/cpp/BUILD b/tools/cpp/BUILD
index 4578c25c9..802d9a914 100644
--- a/tools/cpp/BUILD
+++ b/tools/cpp/BUILD
@@ -1,5 +1,4 @@
 load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
-load("@aspect_bazel_lib//lib:expand_template.bzl", "expand_template_rule")
 
 package(default_visibility = ["//visibility:public"])
 
@@ -16,32 +15,3 @@ bzl_library(
     name = "toolchain_utils",
     srcs = ["toolchain_utils.bzl"],
 )
-
-filegroup(
-    name = "clang_tidy_deps",
-    srcs = [
-        "@llvm-project//clang:builtin_headers_gen",
-        "@llvm-project//clang-tools-extra/clang-tidy",
-    ],
-)
-
-expand_template_rule(
-    name = "clang_tidy_gen",
-    out = "run_clang_tidy.sh",
-    data = [
-        "@llvm-project//clang:builtin_headers_gen",
-        "@llvm-project//clang-tools-extra/clang-tidy",
-    ],
-    is_executable = True,
-    substitutions = {
-        "@RESOURCES@": "$(execpaths @llvm-project//clang:builtin_headers_gen)",
-        "@CLANG_TIDY@": "$(execpath @llvm-project//clang-tools-extra/clang-tidy)",
-    },
-    template = "run_clang_tidy.sh.in",
-)
-
-sh_binary(
-    name = "clang_tidy",
-    srcs = ["run_clang_tidy.sh"],
-    data = [":clang_tidy_deps"],
-)
