diff --git a/internal/transport/http2_client.go b/internal/transport/http2_client.go
index afcda360..33e4d64a 100644
--- a/internal/transport/http2_client.go
+++ b/internal/transport/http2_client.go
@@ -1712,6 +1712,9 @@ func (t *http2Client) keepalive() {
 				if channelz.IsOn() {
 					atomic.AddInt64(&t.czData.kpCount, 1)
 				}
+				if t.kp.PingCallback != nil {
+					t.kp.PingCallback(t.connectionID)
+				}
 				t.controlBuf.put(p)
 				timeoutLeft = t.kp.Timeout
 				outstandingPing = true
diff --git a/keepalive/keepalive.go b/keepalive/keepalive.go
index 34d31b5e..c582c73f 100644
--- a/keepalive/keepalive.go
+++ b/keepalive/keepalive.go
@@ -43,6 +43,7 @@ type ClientParameters struct {
 	// when there are no active RPCs, Time and Timeout will be ignored and no
 	// keepalive pings will be sent.
 	PermitWithoutStream bool // false by default.
+	PingCallback        func(connectionID uint64)
 }
 
 // ServerParameters is used to set keepalive and max-age parameters on the
