diff --git a/internal/transport/handler_server.go b/internal/transport/handler_server.go
index 17f7a21b..93523d73 100644
--- a/internal/transport/handler_server.go
+++ b/internal/transport/handler_server.go
@@ -476,7 +476,6 @@ func (ht *serverHandlerTransport) IncrMsgSent() {}
 func (ht *serverHandlerTransport) incrMsgRecv() {}

 func (ht *serverHandlerTransport) Drain(string) {
-	panic("Drain() is not implemented")
 }

 // mapRecvMsgError returns the non-nil err into the appropriate
