From 87c9faeabe72e6a4ee625a9badcd6bdaa94704a3 Mon Sep 17 00:00:00 2001
From: Zoey Greer <zoey@buildbuddy.io>
Date: Fri, 6 Feb 2026 18:01:53 -0500
Subject: [PATCH] patch

---
 mem/buffer_pool.go | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/mem/buffer_pool.go b/mem/buffer_pool.go
index e37afdd..7ebae14 100644
--- a/mem/buffer_pool.go
+++ b/mem/buffer_pool.go
@@ -43,9 +43,13 @@ const goPageSize = 4 << 10 // 4KiB. N.B. this must be a power of 2.
 var defaultBufferPoolSizes = []int{
 	256,
 	goPageSize,
-	16 << 10, // 16KB (max HTTP/2 frame size used by gRPC)
-	32 << 10, // 32KB (default buffer size for io.Copy)
-	1 << 20,  // 1MB
+	16 << 10,  // 16KB (max HTTP/2 frame size used by gRPC)
+	32 << 10,  // 32KB (default buffer size for io.Copy)
+	64 << 10,  // 64KB
+	128 << 10, // 128KB
+	256 << 10, // 256KB
+	512 << 10, // 512KB
+	1 << 20,   // 1MB
 }
 
 var defaultBufferPool BufferPool
-- 
2.47.3

