diff --git a/internal/transport/handler_server.go b/internal/transport/handler_server.go
index 7ab3422b..49b1af09 100644
--- a/internal/transport/handler_server.go
+++ b/internal/transport/handler_server.go
@@ -480,7 +480,6 @@ func (ht *serverHandlerTransport) runStream() {
 func (ht *serverHandlerTransport) incrMsgRecv() {}
 
 func (ht *serverHandlerTransport) Drain(string) {
-	panic("Drain() is not implemented")
 }
 
 // mapRecvMsgError returns the non-nil err into the appropriate
