diff --git a/rules.bzl b/rules.bzl
index b4a6843..cbb0315 100644
--- a/rules.bzl
+++ b/rules.bzl
@@ -1,4 +1,6 @@
 load("@rules_proto//proto:defs.bzl", "ProtoInfo")
+load("@rules_proto//proto:proto_common.bzl", proto_toolchains = "toolchains")
+load("@com_google_protobuf//bazel/common:proto_common.bzl", "proto_common")
 
 def _trim_prefix(text, prefix):
     if text.startswith(prefix):
@@ -18,6 +20,13 @@ ProtobufjsInfo = provider(
 )
 
 def _protoc_gen_protobufjs_impl(ctx):
+    if getattr(proto_common, "INCOMPATIBLE_ENABLE_PROTO_TOOLCHAIN_RESOLUTION", False):
+        proto_toolchain = ctx.toolchains["@rules_proto//proto:toolchain_type"]
+        if not proto_toolchain:
+            fail("No proto toolchain found.")
+        protoc_executable = proto_toolchain.proto.proto_compiler
+    else:
+        protoc_executable = ctx.executable._protoc
     gen_import_path = _package_path(ctx) + "/" + ctx.attr.out
 
     # Use strict_imports since we expect all direct proto imports to
@@ -71,7 +80,7 @@ def _protoc_gen_protobufjs_impl(ctx):
         outputs = outputs,
         inputs = inputs,
         tools = [ctx.executable._protoc_gen_protobufjs],
-        executable = ctx.executable._protoc,
+        executable = protoc_executable,
         arguments = args,
 
         # Debug:
@@ -131,12 +140,6 @@ protoc_gen_protobufjs = rule(
             """,
             default = [],
         ),
-        "_protoc": attr.label(
-            default = "@com_google_protobuf//:protoc",
-            executable = True,
-            cfg = "exec",
-            allow_single_file = True,
-        ),
         "_protoc_gen_protobufjs": attr.label(
             default = "@com_github_buildbuddy_io_protoc_gen_protobufjs//:protoc-gen-protobufjs",
             executable = True,
@@ -159,5 +162,13 @@ protoc_gen_protobufjs = rule(
                 "@com_google_protobuf//:wrappers_proto",
             ],
         ),
-    },
+    } | proto_toolchains.if_legacy_toolchain({
+        "_protoc": attr.label(
+            default = "@com_google_protobuf//:protoc",
+            executable = True,
+            cfg = "exec",
+            allow_single_file = True,
+        ),
+    }),
+    toolchains = proto_toolchains.use_toolchain("@rules_proto//proto:toolchain_type"),
 )
