diff --git a/include/github.com/planetscale/vtprotobuf/vtproto/BUILD.bazel b/include/github.com/planetscale/vtprotobuf/vtproto/BUILD.bazel
index 418af2e..f356ce6 100644
--- a/include/github.com/planetscale/vtprotobuf/vtproto/BUILD.bazel
+++ b/include/github.com/planetscale/vtprotobuf/vtproto/BUILD.bazel
@@ -7,6 +7,7 @@ proto_library(
     srcs = ["ext.proto"],
     visibility = ["//visibility:public"],
     deps = ["@com_google_protobuf//:descriptor_proto"],
+	strip_import_prefix = "/include"
 )
 
 go_proto_library(
