diff --git a/language/go/def.bzl b/language/go/def.bzl
index a00e0a6..7f5330b 100644
--- a/language/go/def.bzl
+++ b/language/go/def.bzl
@@ -1 +1,7 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_context")
+load(
+    "@io_bazel_rules_go//go/private:context.bzl",
+    "CGO_ATTRS",
+    "CGO_FRAGMENTS",
+    "CGO_TOOLCHAINS",
+)
@@ -25,6 +31,7 @@ std_package_list = rule(
         "_go_context_data": attr.label(
             default = "@io_bazel_rules_go//:go_context_data",
         ),
-    },
-    toolchains = ["@io_bazel_rules_go//go:toolchain"],
+    } | CGO_ATTRS,
+    fragments = CGO_FRAGMENTS,
+    toolchains = ["@io_bazel_rules_go//go:toolchain"] + CGO_TOOLCHAINS,
 )
