diff --git a/cmd/gazelle/BUILD.bazel b/cmd/gazelle/BUILD.bazel
--- a/cmd/gazelle/BUILD.bazel
+++ b/cmd/gazelle/BUILD.bazel
@@ -1,4 +1,5 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
 load("//:def.bzl", "DEFAULT_LANGUAGES", "gazelle_binary")
-
+exports_files(srcs = glob(["*.go"]))
+
 gazelle_binary(
     name = "gazelle",
     languages = DEFAULT_LANGUAGES,
diff --git a/cmd/gazelle/gazelle.go b/cmd/gazelle/gazelle.go
--- a/cmd/gazelle/main.go
+++ b/cmd/gazelle/main.go
@@ -56,6 +56,10 @@ func (cmd command) String() string {
 }
 
 func main() {
+        Run()
+}
+
+func Run() {
 	log.SetPrefix("gazelle: ")
 	log.SetFlags(0) // don't print timestamps
 
diff --git a/internal/module/BUILD.bazel b/internal/module/BUILD.bazel
--- a/internal/module/BUILD.bazel
+++ b/internal/module/BUILD.bazel
@@ -4,5 +4,5 @@ go_library(
     name = "module",
     srcs = ["module.go"],
     importpath = "github.com/bazelbuild/bazel-gazelle/internal/module",
-    visibility = ["//:__subpackages__"],
+    visibility = ["//visibility:public"],
     deps = [
diff --git a/internal/gazelle_binary.bzl b/internal/gazelle_binary.bzl
--- a/internal/gazelle_binary.bzl
+++ b/internal/gazelle_binary.bzl
@@ -15,3 +15,9 @@ load(
     "go_context",
     "new_go_info",
 )
+load(
+    "@io_bazel_rules_go//go/private:context.bzl",
+    "CGO_ATTRS",
+    "CGO_FRAGMENTS",
+    "CGO_TOOLCHAINS",
+)
@@ -100,13 +106,14 @@ _gazelle_binary_kwargs = {
             allow_empty = False,
         ),
         "_go_context_data": attr.label(default = "@io_bazel_rules_go//:go_context_data"),
         # _stdlib is needed for rules_go versions before v0.23.0. After that,
         # _go_context_data includes a dependency on stdlib.
         "_stdlib": attr.label(default = "@io_bazel_rules_go//:stdlib"),
         "_srcs": attr.label(
             default = "//cmd/gazelle:gazelle_lib",
         ),
-    },
+    } | CGO_ATTRS,
+    "fragments": CGO_FRAGMENTS,
     "executable": True,
-    "toolchains": ["@io_bazel_rules_go//go:toolchain"],
+    "toolchains": ["@io_bazel_rules_go//go:toolchain"] + CGO_TOOLCHAINS,
 }
