diff --git a/src/server.h b/src/server.h
index 164a82271..eeb208785 100644
--- a/src/server.h
+++ b/src/server.h
@@ -1742,7 +1742,7 @@ void forceCommandPropagation(client *c, int flags);
 void preventCommandPropagation(client *c);
 void preventCommandAOF(client *c);
 void preventCommandReplication(client *c);
-int prepareForShutdown();
+int prepareForShutdown(int flags);
 #ifdef __GNUC__
 void serverLog(int level, const char *fmt, ...)
     __attribute__((format(printf, 2, 3)));
