diff --git a/MODULE.bazel b/MODULE.bazel
index 1775465..187e414 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,6 +1,6 @@
 module(
     name = "rules_webtesting",
-    version = "0.4.0",
+    version = "0.2.1-0.20250212231324-7a1c88f61e35",
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
@@ -54,15 +54,7 @@ pip.parse(
 use_repo(pip, "rules_webtesting_py_deps")
 
 ##### Scala dependencies #####
-bazel_dep(name = "rules_scala", repo_name = "io_bazel_rules_scala")
-
-# TODO: Switch to a release version once it's available in BCR.
-# Otherwise, the root module has to declare the same override.
-git_override(
-    module_name = "rules_scala",
-    commit = "219e63983e8e483e66ebf70372969ba227382001",
-    remote = "https://github.com/mbland/rules_scala",
-)
+bazel_dep(name = "rules_scala", repo_name = "io_bazel_rules_scala", version = "7.1.2")
 
 scala_deps = use_extension(
     "@io_bazel_rules_scala//scala/extensions:deps.bzl",
@@ -71,9 +63,6 @@ scala_deps = use_extension(
 scala_deps.settings(
     fetch_sources = True,
 )
-scala_deps.toolchains(
-    scalatest = True,
-)
 
 ##### Browsers dependencies #####
 browser_repositories = use_extension("//web:extension.bzl", "browser_repositories_extension")
